content/en/docs/concepts/services-networking/service.md

```diff
@@ -14,44 +14,68 @@ weight: 10
-replicas.  Those replicas are fungible - frontends do not care which backend
-they use.  While the actual `Pods` that compose the backend set may change, the
-frontend clients should not need to be aware of that or keep track of the list
-of backends themselves.  The `Service` abstraction enables this decoupling.
-
-For Kubernetes-native applications, Kubernetes offers a simple `Endpoints` API
-that is updated whenever the set of `Pods` in a `Service` changes.  For
-non-native applications, Kubernetes offers a virtual-IP-based bridge to Services
-which redirects to the backend `Pods`.
+{{< glossary_definition term_id="service" length="short" >}}
+
+No need to modify your application to use an unfamiliar service discovery mechanism.
+Kubernetes gives pods their own IP addresses and a single DNS name for a set of pods,
+and can load-balance across them.
 
 {{% /capture %}}
 
 {{% capture body %}}
 
+## Motivation
+
+Kubernetes {{< glossary_tooltip term_id="pod" text="Pods" >}}  are mortal.
+They are born and when they die, they are not resurrected.
+If you use a {{< glossary_tooltip term_id="deployment" >}} to run your app,
+it can create and destroy Pods dynamically (e.g. when scaling out or in).
+
+Each Pod gets its own IP address, however the set of Pods
+for a Deployment running in one moment in time could be different from
+the set of Pods running that application a moment later.
+
+This leads to a problem: if some set of Pods (call them “backends”) provides
+functionality to other Pods (call them “frontends”) inside your cluster,
+how do those frontends find out and keep track of which IP address to connect
+to, so that the frontend can use the backend part of the workload?
+
+Enter _Services_.
+
+## Service resources {#service-resource}
+
+In Kubernetes, a Service is an abstraction which defines a logical set of Pods
+and a policy by which to access them (you'll sometimes see this pattern called
+a micro-service). The set of Pods targeted by a Service is usually determined
+by a {{< glossary_tooltip text="selector" term_id="selector" >}}
+(see [below](#services-without-selectors) for why you might want a Service
+_without_ a selector).
+
+For example: consider a stateless image-processing backend which is running with
+3 replicas.  Those replicas are fungible&mdash;frontends do not care which backend
+they use.  While the actual Pods that compose the backend set may change, the
+frontend clients should not need to be aware of that, nor should they need to keep
+track of the set of backends themselves.
+
+The Service abstraction enables this decoupling.
+
+### Cloud-native service discovery
+
+If you're able to use Kubernetes APIs for service discovery in your application,
+you can query the {{< glossary_tooltip text="API server" term_id="kube-apiserver" >}}
+for Endpoints, that will be updated whenever the set of Pods in a Service changes.
+
+For non-native applications, Kubernetes offers ways to place a network port or load
```

Comment:
nit: replace `balancer to between` -> `balancer in between`

Reply:
Thanks for spotting.